spec/lib/task-spec.js

```diff
@@ -508,6 +502,64 @@ describe("Task", function () {
 
     });
 
+    describe("timer", function() {
+        it("should not timeout when value is 0", function() {
+            var definition = _.cloneDeep(noopDefinition);
+            definition.options.timeout = 0;
+            var task = Task.create(definition, {}, {});
+            task.renderAll = sinon.stub().resolves();
+            sinon.spy(task, 'cancel');
+
+            return task.run().then(function() {
+                expect(task.state).to.equal('succeeded');
+                expect(task.error).to.equal(null);
+            });
+        });
+
+        it("should not timeout when value is -1", function() {
+            var definition = _.cloneDeep(noopDefinition);
+            definition.options.timeout = -1;
+            var task = Task.create(definition, {}, {});
+            task.renderAll = sinon.stub().resolves();
+            sinon.spy(task, 'cancel');
+
+            return task.run().then(function() {
+                expect(task.state).to.equal('succeeded');
+                expect(task.error).to.equal(null);
+            });
+        });
+
+        it("should timeout with options.timeout", function() {
```

Comment:
@yyscamper @jlongever I am using real delays, but not any that involve I/O. `Promise.delay` eventually gets called here: https://github.com/RackHD/on-tasks/blob/master/lib/jobs/noop-job.js#L37. I'm taking advantage of the fact that even during normal execution the setTimeout of 1ms will trigger before the job is started and completes (since `Promise.delay(0)` isn't synchonous). However, this could be more deterministic, so I'll update the test task definition to add a 2ms delay to the job, if that sounds alright.